core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1445,10 +1444,7 @@ void persistMetadata(ShardRouting newRouting, ShardRouting currentRouting) {
     }
 
     private String getIndexUUID() {
-        assert indexSettings.get(IndexMetaData.SETTING_INDEX_UUID) != null
-                || indexSettings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).before(Version.V_0_90_6) :
```

Comment:
This assertion didn't make it into IndexSettings class. I know that compatibility to V_0_90_6 is irrelevant but the first part of the assertion might still be interesting?